lib/ProMotion/collection/collection.rb

```diff
@@ -0,0 +1,106 @@
+module ProMotion
+  module Collection
+    include ProMotion::Styling
+    include ProMotion::Table::Utils
+    include ProMotion::CollectionBuilder
+
+    attr_reader :promotion_collection_data
+
+    def collection_view
+      self.collectionView
+    end
+
+    def screen_setup
+      check_collection_data
+    end
+
+    # Override viewDidLoad here to register cell classes here.
+    # Calling it from `screen_setup` cause `on_load` to be called twice
```

Comment:
@jamonholmgren @bmichotte There seems to be some weirdness with `on_load` still. I'm unable to set the background color on the `collection_view` and `set_nav_bar_button` doesn't work either. Moving both of these to `set_up_register_class` does work though. Maybe something to do with this override?